### PR TITLE
Shrink multi_head_attention_forward grad OpInfo sample

### LINKED ISSUE #4835: [UT] Test case runs more than 7 hours on CI windows runner
### 🐛 Describe the bug

Please refer https://github.com/intel/torch-xpu-ops/actions/runs/31014497073/job/92334961094#step:9:4600 which runs on a MTL Windows node

```
Wed, 05 Aug 2026 14:39:06 GMT
test_ops_xpu.py::TestCompositeComplianceXPU::test_forward_ad_nn_functional_mse_loss_xpu_float32 PASSED [ 86%]
Wed, 05 Aug 2026 22:10:18 GMT
test_ops_xpu.py::TestCompositeComplianceXPU::test_forward_ad_nn_functional_multi_head_attention_forward_xpu_float32 PASSED [ 86%]
```

Failed on BMG Linux
Cases:
op_extended,third_party.torch-xpu-ops.test.xpu.extended.test_ops_xpu.TestCompositeComplianceXPU,test_forward_ad_nn_functional_multi_head_attention_forward_xpu_float32

### Versions

main

### PR DESCRIPTION
**Pull request by PatrykWilczewski**
The composite-compliance forward-AD check
(composite_compliance.check_forward_ad_formula) wraps every differentiable tensor argument in a tensor subclass at two nesting levels (args x tangent args), so it's cost grows as ~4^(num tensor args). The requires_grad OpInfo sample for
nn.functional.multi_head_attention_forward carried 10 tensor args (q, k, v, in_proj_weight, in/out proj biases, bias_k, bias_v, and a 2d attn_mask), giving ~4^10 = 1.048.576 forward executions for a single sample. On slow integrated GPUs this ran for over 7 hours

Fix: reduce the requires_grad sample to 5 differentiable tensors by dropping the attention mask (mask_types ("2d",) -> (None,)) and keeping the in/out projection biases and bias_k/bias_v as None. This cuts the forward-AD cost to ~4^5 = 1.024 executions while still exercising the biassless MHA forward-AD path.

Fixes: https://github.com/intel/torch-xpu-ops/issues/4835

### CODE REVIEW

**Review by dominikjastrzebskix — APPROVED** [review]

**Review by BBBela — COMMENTED** [review]
This test case `test_forward_ad_nn_functional_multi_head_attention_forward` is skipped in PyTorch main `test_ops.py` test file (probably also because it takes too long). We launch it in the `op_extended` suite only as something additional. I think we can just skip it on XPU instead of changing the upstream test behavior for all backends. That will be simpler and much faster to merge.

I know it is a round trip as the original PR did exactly that, but still, I think the original solution with skip is the right one at this moment. Although I would add one general skip directly in the `extended/test_ops_xpu.py` test file in the `@skipOps()` decorator just above the `test_forward_ad()` test function instead of adding to the `skip_list_common.py`
